GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerTestCase/DiskLayoutUpgradeTests.cs

```diff
@@ -0,0 +1,131 @@
+
+            string target = Path.Combine(this.Enlistment.EnlistmentRoot, "GVFS", "GVFS", "GVFS.UnitTests");
+            string link = Path.Combine(this.Enlistment.RepoRoot, "UnitTests");
+            ProcessHelper.Run("CMD.exe", "/C mklink /J " + link + " " + target);
+            target = Path.Combine(this.Enlistment.EnlistmentRoot, "GVFS", "GVFS", "GVFS.Installer");
+            link = Path.Combine(this.Enlistment.RepoRoot, "Installer");
+            ProcessHelper.Run("CMD.exe", "/C mklink /D " + link + " " + target);
+        }
+
+        protected void PlaceholderDatabaseShouldIncludeCommonLinesForUpgradeTestIO(string[] placeholderLines)
+        {
+            placeholderLines.ShouldContain(x => x.Contains("A Readme.md"));
+            placeholderLines.ShouldContain(x => x.Contains("A Scripts\\RunUnitTests.bat"));
+            placeholderLines.ShouldContain(x => x.Contains("A GVFS\\GVFS.Common\\Git\\GitRefs.cs"));
+            placeholderLines.ShouldContain(x => x.Contains("A .gitignore"));
+            placeholderLines.ShouldContain(x => x == "A Scripts\0" + TestConstants.PartialFolderPlaceholderDatabaseValue);
+            placeholderLines.ShouldContain(x => x == "A GVFS\0" + TestConstants.PartialFolderPlaceholderDatabaseValue);
+            placeholderLines.ShouldContain(x => x == "A GVFS\\GVFS.Common\0" + TestConstants.PartialFolderPlaceholderDatabaseValue);
+            placeholderLines.ShouldContain(x => x == "A GVFS\\GVFS.Common\\Git\0" + TestConstants.PartialFolderPlaceholderDatabaseValue);
+            placeholderLines.ShouldContain(x => x == "A GVFS\\GVFS.Tests\0" + TestConstants.PartialFolderPlaceholderDatabaseValue);
+        }
+
+        protected string[] GetPlaceholderDatabaseLinesBeforeUpgrade(string placeholderDatabasePath)
+        {
+            placeholderDatabasePath.ShouldBeAFile(this.fileSystem);
+            string[] lines = this.fileSystem.ReadAllText(placeholderDatabasePath).Split(new char[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
+            lines.Length.ShouldEqual(12);
+            this.PlaceholderDatabaseShouldIncludeCommonLinesForUpgradeTestIO(lines);
+            lines.ShouldContain(x => x.Contains("A GVFS\\GVFS.Tests\\Properties\\AssemblyInfo.cs"));
+            lines.ShouldContain(x => x == "D GVFS\\GVFS.Tests\\Properties\\AssemblyInfo.cs");
+            lines.ShouldContain(x => x == "A GVFS\\GVFS.Tests\\Properties\0" + TestConstants.PartialFolderPlaceholderDatabaseValue);
+            return lines;
+        }
+
+        protected string FilePlaceholderString(params string[] pathParts)
+        {
+            return $"{Path.Combine(pathParts)}{GVFSHelpers.PlaceholderFieldDelimiter}0{GVFSHelpers.PlaceholderFieldDelimiter}";
+        }
+
+        protected string PartialFolderPlaceholderString(params string[] pathParts)
+        {
+            return $"{Path.Combine(pathParts)}{GVFSHelpers.PlaceholderFieldDelimiter}1{GVFSHelpers.PlaceholderFieldDelimiter}";

```

Comment:
Minor: A named constant here for `1` would make it easier to understand why it's part of the string